src/util/chunk_fs.js

```diff
@@ -44,8 +44,10 @@ class ChunkFS extends stream.Transform {
                 const buf = (available_size < chunk.length) ? chunk.slice(0, available_size) : chunk;
                 this.q_buffers.push(buf);
                 this.q_size += buf.length;
-                // Should flush when equals, but added greater than just in case
-                if (this.q_size >= config.NSFS_BUF_SIZE) await this._flush_buffers();
+                // Should flush when num of chunks equals to max iov which is the limit according to https://linux.die.net/man/2/writev
+                // or when q_size equals to config.NSFS_BUF_SIZE, but added greater than just in case
+                const iov_max = nb_native().fs.PLATFORM_IOV_MAX || config.NSFS_DEFAULT_IOV_MAX;
```

Comment:
How about taking this as the minimum of the config and the platform's value? Just need to make sure that these values are not undefined.
Also I would move this to the ctor instead of per buffer.
```suggestion
                const iov_max = Math.min(nb_native().fs.PLATFORM_IOV_MAX, config.NSFS_DEFAULT_IOV_MAX);
```

Reply:
Sounds goood :)